Golem.Tests/Utils.cs

```diff
@@ -100,25 +100,49 @@ public static async Task WaitForProviderPidFile(string golemPath, int timeoutSec
         /// Reads from `channel` and returns first `T` for which `matcher` returns `false`
         /// </summary>
         /// <exception cref="Exception">Thrown when reading channel exceeds in total `timeoutMs`</exception>
-        public async static Task<T> ReadChannel<T>(ChannelReader<T> channel, Func<T, bool>? matcher = null, double timeoutMs = 10_000, ILogger? logger = null)
+        public async static Task<T> ReadChannel<T>(ChannelReader<T> channel, Func<T, bool>? filter = null, TimeSpan? timeout = null, ILogger? logger = null)
         {
+            var timeout_ = timeout ?? TimeSpan.FromSeconds(10);
```

Comment:
this is a really bad name for the timeout. What is the difference between `timeout_` and `timeout`?